src/velocity_smoother_node.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2020, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of Yujin Robot nor the names of its
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <cstdio>
+#include <memory>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "velocity_smoother/velocity_smoother.hpp"
+
+int main(int argc, char** argv)
```

Comment:
First experiences with `rclcpp` for me and this caught me by surprise. I would have thought you could have started up a node without needing to create an executable - bit like a standalone nodelet in ROS1.

Reply:
There are a few things to note here.

1.  As of Dashing, there is a composable container that you can always load nodes into; you can see it in use in the `velocity_smoother-composed-launch.py` launch file.
1.  However, for simple uses, the composed node is usually overkill, so I like to provide a standalone "node" as well.
1.  As of Eloquent, we have a CMake macro that will auto-generate code like this so that you don't have to write this boilerplate.
1.  But that macro doesn't exist in Dashing.

That chain of logic leads me to always including this, particularly for things that we could release into Dashing (I think this qualifies).  It's not absolutely essential, but I think it is a nice-to-have.

Reply:
Understood and concur!